File: kittens/src/main/java/com/example/kittens/utils/StateBinder.java
package com.example.kittens.utils;

import android.app.Activity;
import android.view.View;

import se.snylt.witch.viewbinder.Witch;

public class StateBinder<State> {

    private final Object viewFinder;

    public StateBinder(Object viewFinder) {
        this.viewFinder = viewFinder;
    }

    public void bind(State target) {
        if(viewFinder instanceof Activity) {
            Witch.bind(target, (Activity) viewFinder);
        } else {
            Witch.bind(target, (View) viewFinder);
        }
    }
}

File: todo/src/main/java/com/example/todo/utils/StateBinder.kt
package com.example.todo.utils

import android.app.Activity
import android.util.Log
import android.view.View
import se.snylt.witch.viewbinder.Witch

class StateBinder<in State> (val viewFinder: Any) {

  fun bind(state: State) {
    if (viewFinder is Activity) {
      Witch.bind(state, viewFinder)
    } else if (viewFinder is View) {
      Witch.bind(state, viewFinder)
    }
    Log.d("State", state.toString())
  }
}


